src/constructors/createGlobalStyle.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import React from 'react'
+import hashStr from '../vendor/glamor/hash'
+import StyleSheet from '../models/StyleSheet'
+import type { Interpolation, Stringifier } from '../types'
+
+const _createGlobalStyle = (stringifyRules: Stringifier, css: Function) => {
+  const createGlobalStyle = (strings: Array<string>, ...interpolations: Array<Interpolation>) => {
+    return class extends React.Component {
+      componentId = ''
+
+      componentDidMount() {
```

Comment:
Here the component should use `componentWillMount` instead :wink: otherwise this wouldn't work correctly for SSR. It's also important as we'll want the global styles to apply immediately before all DOM nodes mount (that might be referred to in or affected by these global styles).

Reply:
Thanks! Adjusted in https://github.com/styled-components/styled-components/pull/1416/commits/df87dd75044e19144a80ca6403cae475dce155d7